flyway-core/src/main/java/org/flywaydb/core/internal/dbsupport/oracle/OracleSqlStatementBuilder.java

```diff
@@ -118,6 +118,13 @@ protected String computeAlternateCloseQuote(String openQuote) {
 
     @Override
     public boolean canDiscard() {
-        return super.canDiscard() || statementStart.startsWith("SET DEFINE OFF");
+        return super.canDiscard() || isSqlPlusCommand();
+    }
+
+    private boolean isSqlPlusCommand() {
+        return statementStart.matches("SET\\s+(DEFINE|ECHO|TIMING|SERVEROUTPUT)\\s+(ON|OFF).*")
```

Comment:
Please add "SHOW ERRORS" as well in this list

Reply:
We can also ignore `prompt` (or even better, log the statement behind it.
And also ignore `spool`